plugins/BEdita/Core/src/Model/Schema/JsonSchema.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Schema;
+
+use Cake\Datasource\Exception\RecordNotFoundException;
+use Cake\Network\Exception\NotFoundException;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Inflector;
+
+/**
+ * JSON Schema generation utilities
+ *
+ * Provides static methods to generate JSON Schema for objects and resources
+ */
+class JsonSchema
+{
+    /**
+     * Valid resource names
+     *
+     * @var array
+     */
+    const VALID_RESOURCES = [
+        'applications',
+        'roles',
+        'streams',
+    ];
+
+    /**
+     * Generate JSON Schema draft-6 (https://tools.ietf.org/html/draft-wright-json-schema-01)
+     *
+     * @param string $typeName Type name of a resource or object
+     * @param string $url Url of this schema
+     * @return array Associative array representing schema
+     */
+    public static function generate($typeName, $url)
+    {
+        $isResource = in_array($typeName, static::VALID_RESOURCES);
+        $schema = [];
```

Comment:
This assignement is useless: it is always immediately overwritten, so this line can be removed.